src/preset-themes/jsonSchema/jsonSchema.js

```diff
@@ -0,0 +1,227 @@
+                        "h": {
+                            "type": "number",
+                            "maximum": 360,
+                            "minimum": 0
+                        },
+                        "s": {
+                            "type": "number",
+                            "maximum": 100,
+                            "minimum": 0
+                        },
+                        "l": {
+                            "type": "number",
+                            "maximum": 100,
+                            "minimum": 0
+                        },
+                        "a": {
+                            "type": "number",
+                            "maximum": 100,
+                            "minimum": 0
+                        }
+                    },
+                    "required": ["h", "s", "l"]
+                },
+                "toolbar_field_text": {
+                    "type": "object",
+                    "properties": {
+                        "h": {
+                            "type": "number",
+                            "maximum": 360,
+                            "minimum": 0
+                        },
+                        "s": {
+                            "type": "number",
+                            "maximum": 100,
+                            "minimum": 0
+                        },
+                        "l": {
+                            "type": "number",
+                            "maximum": 100,
+                            "minimum": 0
+                        },
+                        "a": {
+                            "type": "number",
+                            "maximum": 100,
+                            "minimum": 0
+                        }
+                    },
+                    "required": ["h", "s", "l"]
+                }
+            },
+            "required": ["toolbar", "toolbar_text", "accentcolor", "textcolor", "toolbar_field", "toolbar_field_text"]
+        }
+    }
+};
+
+var validate = ajv.compile(schema);
+
+require("fs").readFile("../001.json", "utf8", function(err, data) {
+    if (err) {
+        log(err);
+    } else {
+        var obj1 = JSON.parse(data);
+        test(obj1);
+    }
+});
+require("fs").readFile("../002.json", "utf8", function(err, data) {
+    if (err) {
+        log(err);
+    } else {
+        var obj2 = JSON.parse(data);
+        test(obj2);
+    }
+});
+require("fs").readFile("../003.json", "utf8", function(err, data) {
+    if (err) {
+        log(err);
+    } else {
+        var obj3 = JSON.parse(data);
+        test(obj3);
+    }
+});
+require("fs").readFile("../004.json", "utf8", function(err, data) {
+    if (err) {
+        log(err);
+    } else {
+        var obj4 = JSON.parse(data);
+        test(obj4);
+    }
+});
+require("fs").readFile("../005.json", "utf8", function(err, data) {
+    if (err) {
+        log(err);
+    } else {
+        var obj5 = JSON.parse(data);
+        test(obj5);
+    }
+});
```

Comment:
Instead of `fs.readFile()` and parsing as JSON, one approach (not great, but it's _an_ approach) would be to require the JSON file using a relative path:

```js
const validate = ajv.compile(schema);

const theme001 = require("../001.json");
test(theme001);

const theme002 = require("../002.json");
test(theme002);

const theme003 = require("../003.json");
test(theme003);

const theme004 = require("../004.json");
test(theme004);

const theme005 = require("../005.json");
test(theme005);

function test(data) {
  const valid = validate(data);
  if (valid) {
    console.log("Valid!");
  } else {
    console.log("Invalid: " + ajv.errorsText(validate.errors));
  }
}
```

Still not great, since you'd have to declare each file specifically, and you're currently only linting 5 of the 7 themes in that directory.

Another possible option is to use `fs.readdir()` or `fs.readdirSync()` and do something like this:

```js
const validate = ajv.compile(schema);

fs.readdirSync("./src/preset-themes")
  .filter(file => file.endsWith(".json"))
  .forEach(file => test(file));

function test(file) {
  const data = require(`../${file}`);
  const valid = validate(data);
  if (valid) {
    console.log(`${file}: Valid!`);
  } else {
    console.log(`${file}: Invalid: ${ajv.errorsText(validate.errors)}`);
    process.exitCode = 1;
  }
}
```

... which gives me the following output:

```sh
➜  ThemesRFun git:(PR172) ✗ node ./src/preset-themes/jsonSchema/jsonSchema.js
001.json: Valid!
002.json: Valid!
003.json: Valid!
004.json: Valid!
005.json: Valid!
default.json: Valid!
hotdog.json: Valid!
```

Not sure if it works 100%, we'd have to test it with some theme errors, but may get you a bit closer.

Final note is that everything seems to lint as valid, so I think we're still missing some pieces, as theme 005 is still invalid (missing `title`, bonus `title` property in the wrong place, schema isn't complaining we accept `a` fields where we shouldn't, and some themes may have an `images` object which has a `headerURL`:

https://github.com/mozilla/Themer/blob/38933c93a6c304a70b095dfd737b3b2c8ce871d6/src/preset-themes/005.json#L37-L39

Or an `images` object which has an `additional_backgrounds` array with 0-1 items:

https://github.com/mozilla/Themer/blob/38933c93a6c304a70b095dfd737b3b2c8ce871d6/src/preset-themes/default.json#L37-L39

But overall I think this is pretty close!

Reply:
@pdehaan Thank you so much!
I haven't set `additionalProperties: false` to any scope. So we can add additional objects like `"images": { 
   "headerURL": "./bg-016.svg" 
 } `

Also, for theme 005, lint should be invalid because we are interchanging `title` (which is actually of type `string` ) and `colors` (which is actually of type `object`).  When I checked it online, it was giving error like `Invalid: data.title should be string, data.colors should be object`, which is quite evident. I will check it once again and let you know the results. :-)

Thanks!